### DISCUSSION TITLE
I have two questions about ROS2 with DDS-Security.

### DISCUSSION THREAD

**Discussion by dhkstnrh** [eProsima/Fast-DDS#3969] [Q&A]
Hello!

I'm a student studying ROS2.

I have two questions while studying DDS-Security.

First, DDS-Security v1.1 document states, "Secure DDS systems need to co-exist in the same network and, in some cases, interoperate with non-secure DDS systems."
How does ROS2 with DDS-Security interoperate with regular ROS2 that doesn't have DDS-Security applied? ROS2 with DDS-Security communicates between Pub/Subs based on certificates, so I don't understand how it interoperates with regular ROS2 where no certificates exist.

Secondly, is it absolutely necessary for ROS2 with DDS-Security to interoperate with ROS2 without DDS-Security? Can't they be used separately? I'm curious about real-world use cases.

**Comment by bl4ckhyun** [comment]
yoshi! i wanna know about it too!!

**Comment by JLBuenoLopez** [comment]
Hi @dhkstnrh 

DDS Security is configured using several security files as you state in your question. But the parameters you can configure allows for several communication models. Specifically, the [Security Access Control plugin](https://fast-dds.docs.eprosima.com/en/latest/fastdds/security/access_control_plugin/access_control_plugin.html#access-control-plugin-dds-access-permissions) configures which DomainParticipants and Endpoints can access the secure communication by means of two security files: Governance and Permissions. One option configurable in the Governance file is to [allow into the secure communication unauthenticated Participants](https://fast-dds.docs.eprosima.com/en/latest/fastdds/security/access_control_plugin/access_control_plugin.html#allow-unauthenticated-participants), so a non-secure DomainParticipant would be able to communicate with a secure one.

**Reply by dhkstnrh** [reply]
Thank you for your answer!
